staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -700,9 +702,9 @@ func TestCacherNoLeakWithMultipleWatchers(t *testing.T) {
 
 	// wait out the expiration period and pop expired watchers
 	time.Sleep(2 * time.Second)
-	cacher.bookmarkWatchers.popExpiredWatchers()
-	cacher.bookmarkWatchers.lock.Lock()
-	defer cacher.bookmarkWatchers.lock.Unlock()
+	cacher.Lock()
+	cacher.bookmarkWatchers.popExpiredWatchersThreadUnsafe()
+	cacher.Unlock()
```

Comment:
Why did you change this line?
Below we're also accessing bookmarkWatchers, so we should (as previously) do that under lock.

Reply:
You didn't understand my comment - we need to aquire a lock.

I'm just saying you shouldn't unlock here, just revert back to "defer unlock"

Reply:
ah, now I see it, `popExpiredWatchersThreadUnsafe` also accesses the `bookmarkWatchers`, good catch, thx!